### PR TITLE
Remove the Santa kernel extension.

### PR DESCRIPTION
**Pull request by tnek**
This PR removes:
* All of the code in `Source/santa_driver` containing the kernel extension
* The `SNTDriverManager` event provider
* All workflows in our CI related to testing if the driver builds
* Installation of the driver in `install.sh`. Note that code uninstalling existing instances of the driver is still intentionally kept present.
* Kernel extension-specific build rules
* Renames `SNTKernelCommon` to `SNTCommon`
* Driver version output from `santactl version`
* The `[SNTConfigurator EnableSystemExtension] configuration key`

### CODE REVIEW

File: Source/santactl/Commands/SNTCommandVersion.m [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #import "Source/common/SNTCommonEnums.h"
 #import "Source/common/SNTConfigurator.h"
 #import "Source/common/SNTFileInfo.h"
-#import "Source/common/SNTKernelCommon.h"
+#import "Source/common/SNTCommon.h"
 #import "Source/santactl/SNTCommand.h"
 #import "Source/santactl/SNTCommandController.h"
```
Comment (russellhancox) [inline Source/santactl/Commands/SNTCommandVersion.m:25]: Also remove the driver version field from the output

**Review by russellhancox — CHANGES_REQUESTED** [review]

File: Source/santactl/Commands/SNTCommandVersion.m (hunk 1 above)
Comment (tnek) [reply Source/santactl/Commands/SNTCommandVersion.m:24]: Done

**Review by russellhancox — DISMISSED** [review]

**Comment by tnek** [comment]
Caught a few other references to SNTDriverManager to remove that was breaking tests. PTAL, this required rewriting parts of a few unit tests to mock an abstract implementation of the EventProvider protocol instead the more concrete SNTDriverManager class.

File: Source/santactl/BUILD [hunk 2]
```diff
@@ -5,7 +5,7 @@ licenses(["notice"])
 
 package(
     default_visibility = ["//:santa_package_group"],
-    features = ["-layering_check"],
+    features = ["layering_check"],
```
Comment (mlw) [inline Source/santactl/BUILD:8]: This is the `fix.sh` script adding this back in

File: Source/common/BUILD [hunk 3]
```diff
@@ -3,7 +3,7 @@ load("@rules_proto_grpc//objc:defs.bzl", "objc_proto_library")
 
 package(
     default_visibility = ["//:santa_package_group"],
-    features = ["-layering_check"],
+    features = ["layering_check"],
```
Comment (mlw) [inline Source/common/BUILD:6]: Changed by `fix.sh`

File: Source/santad/SNTApplication.m [hunk 4]
```diff
@@ -61,19 +60,12 @@ - (instancetype)init {
   if (self) {
     SNTConfigurator *configurator = [SNTConfigurator configurator];
 
-    // Choose an event logger.
-    // Locate and connect to driver / SystemExtension
-    if ([configurator enableSystemExtension]) {
```
Comment (russellhancox) [inline Source/santad/SNTApplication.m:66]: Might as well remove this key from SNTConfigurator.{h,m} and any documentation too

File: Source/santad/main.m [hunk 5]
```diff
@@ -94,7 +93,6 @@ void cleanup() {
   LOGI(@"com.google.santa.daemon is running from an unexpected path: cleaning up");
   NSFileManager *fm = [NSFileManager defaultManager];
   [fm removeItemAtPath:@"/Library/LaunchDaemons/com.google.santad.plist" error:NULL];
-  [SNTDriverManager unloadDriver];
   [fm removeItemAtPath:@"/Library/Extensions/santa-driver.kext" error:NULL];
```
Comment (russellhancox) [inline Source/santad/main.m:96]: Might as well remove this line too - if we're not going to try unloading

File: Source/santad/SNTApplication.m (hunk 4 above)
Comment (tnek) [reply Source/santad/SNTApplication.m:66]: Done. Surprisingly, I don't see the EnableSystemExtension key itself mentioned anywhere in the documentation.

File: Source/santad/main.m (hunk 5 above)
Comment (tnek) [reply Source/santad/main.m]: Done.

**Review by russellhancox — APPROVED** [review]

**Comment by pmarkowsky** [comment]
This is the bulk of the work for #738. 

Still need to remove the enums.

**Pull request merged**